tests/ballerina-compiler-api-test/src/test/java/io/ballerina/semantic/api/test/symbols/TypeSymbolTest.java

```diff
@@ -78,48 +80,48 @@ public void setup() {
     }
 
     @Test(dataProvider = "PosProvider")
-    public void testTypeSymbolLookup(int line, int col, TypeDescKind typeKind, String signature) {
+    public void testTypeSymbolLookup(int line, int col, TypeDescKind typeKind, String signature, boolean hasLocation) {
         TypeSymbol type = assertBasicsAndGetType(line, col, typeKind, signature);
         assertTrue(type.getName().isEmpty());
-        assertTrue(type.getLocation().isEmpty());
+        assertEquals(type.getLocation().isPresent(), hasLocation);
```

Comment:
Instead of checking has-location shall we check the location values? For the items that doesn't have locations, we can use null or default-location (if applicable)

Reply:
We may also combine the new test with this one WDYT?